src/cli/cli.cc

```diff
@@ -70,6 +70,66 @@ static const std::unordered_map<InstallResult::Status, StatusCode> i2s = {
+     "(offline mode only)"},
+    {SC::CheckinSecurityError, "Invalid TUF metadata"},
+    {SC::CheckinExpiredMetadata, "TUF metadata is expired"},
+    {SC::CheckinMetadataFetchFailure, "Unable to fetch TUF metadata"},
+    {SC::CheckinMetadataNotFound, "TUF metadata not found in the provided path (offline mode only)"},
+    {SC::CheckinInvalidBundleMetadata,
+     "The bundle metadata is invalid (offline mode only)."
+     "There are a few reasons why the metadata might be invalid:\n"
+     "        1. One or more bundle signatures is/are invalid.\n"
+     "        2. The bundle targets' type, whether CI or production, differs from the device's type.\n"
+     "        3. The bundle targets' tag differs from the device's tag."},
+    {SC::TufTargetNotFound, "Selected target not found"},
+    {SC::CheckinUpdateNewVersion, "Update required: new version"},
+    {SC::CheckinUpdateSyncApps, "Update required: apps synchronization"},
+    {SC::CheckinUpdateRollback, "Update required: rollback"},
+
+    // Return codes for Pull and Install
+    {SC::RollbackTargetNotFound,
+     "Unable to find target to rollback to after a failure to start Apps at boot on a new version of sysroot"},
+    {SC::InstallationInProgress, "Unable to pull/install: there is an installation that needs completion"},
+    {SC::DownloadFailure, "Unable to download target"},
+    {SC::DownloadFailureNoSpace, "There is no enough free space to download the target"},
+    {SC::DownloadFailureVerificationFailed,
+     "The pulled target content is invalid, specifically App compose file is invalid"},
+    {SC::InstallAlreadyInstalled, "Selected target is already installed"},
+    {SC::InstallDowngradeAttempt, "Attempted to install a previous version"},
+
+    // Return codes for Install
+    {SC::InstallAppsNeedFinalization, "Execute the `run` subcommand to finalize installation"},
+    {SC::InstallAppPullFailure, "Unable read target data, make sure it was pulled"},
+    {SC::InstallNeedsRebootForBootFw,
+     "Reboot is required to complete the previous boot firmware update. After reboot the update attempt must be "
+     "repeated from the beginning"},
+
+    // Return codes for Install and CompleteInstall
+    {SC::InstallNeedsReboot, "Reboot to finalize installation"},
+    {SC::OkNeedsRebootForBootFw, "Reboot to finalize bootloader installation"},
+    {SC::InstallRollbackNeedsReboot, "Installation failed, rollback initiated but requires reboot to finalize"},
+
+    // Return codes for CompleteInstall
+    {SC::NoPendingInstallation, "No pending installation to run"},
+    {SC::InstallOfflineRollbackOk, "Offline installation failed, rollback performed"},
+    {SC::InstallRollbackOk, "Online installation failed, rollback performed"},
+    {SC::InstallRollbackFailed, "Installation failed and rollback operation was not successful"},
+    {SC::UnknownError, "Unknown error"},
+};
+
+std::string StatusCodeDescription(StatusCode status) { return status2string.at(status); }
```

Comment:
I think we need to check whether the key (`status`) exists in the map, otherwise the `.at` call will issue an exception. Please, check the code at line 15, I think it is worth doing similar here.

Reply:
Done